gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -145,24 +145,36 @@ impl ClnRpcService {
             let socket = PathBuf::from(config.lightning_dir).join(config.rpc_file);
             let client = ClnRpc::new(socket).await.expect("connect to ln_socket");
 
-            let addr = match plugin.option("addr") {
-                Some(options::Value::String(address)) => SocketAddr::from_str(&address)
-                    .map_err(|e| ClnExtensionError::Error(anyhow!("{}", e)))?,
-                _ => unreachable!(),
-            };
-
             Ok(Self {
                 client: Arc::new(Mutex::new(client)),
                 task_group: TaskGroup::new(),
                 interceptor,
-                addr,
+                listen: plugin.option("listen"),
             })
         } else {
             Err(ClnExtensionError::Error(anyhow!(
                 "Failed to start cln plugin"
             )))
         }
     }
+
+    pub fn get_listen_address(&self) -> Result<SocketAddr, ClnExtensionError> {
+        match self.listen.clone() {
+            Some(options::Value::String(listen)) => {
+                if &listen == "default-dont-use" {
+                    Err(ClnExtensionError::Error(anyhow!(
+                        "listen address option missing on gateway cln extension"
```

Comment:
This error should indicate is which ways the bind address can be provided: as env var or in CLN cfg